--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/SBLList.securitysage	2008-10-05 12:32:41.000000000 -0600
+++ smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/SBLList	2008-10-05 12:32:59.000000000 -0600
@@ -1 +1 @@
-bogusmx.rfc-ignorant.org:multi.surbl.org:black.uribl.com:rhsbl.sorbs.net:bulk.rhs.mailpolice.com:fraud.rhs.mailpolice.com:porn.rhs.mailpolice.com:adult.rhs.mailpolice.com:ex.dnsbl.org:blackhole.securitysage.com
+bogusmx.rfc-ignorant.org:multi.surbl.org:black.uribl.com:rhsbl.sorbs.net:bulk.rhs.mailpolice.com:fraud.rhs.mailpolice.com:porn.rhs.mailpolice.com:adult.rhs.mailpolice.com:ex.dnsbl.org
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/migrate/20securitysage.securitysage	2008-10-05 12:36:50.000000000 -0600
+++ smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/migrate/20securitysage	2008-10-05 12:39:06.000000000 -0600
@@ -0,0 +1,8 @@
+{
+    $qpsmtpd = $DB->get('qpsmtpd');
+    return unless $qpsmtpd;
+    return unless $qpsmtpd->prop('SBLList');
+
+    @sbl = grep { !/blackhole.securitysage.com$/ } split /[,:]/, $qpsmtpd->prop('SBLList');
+    $qpsmtpd->set_prop('SBLList', join ':', @sbl);
+}
