diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/qmail/control/badmailfrom/10sample mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/qmail/control/badmailfrom/10sample
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/qmail/control/badmailfrom/10sample	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/qmail/control/badmailfrom/10sample	2005-04-20 21:19:07.000000000 -0600
@@ -0,0 +1,2 @@
+#spamer@spamdomain.com
+#@spamdomain.com
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/qmail/control/badmailfrom/template-begin mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/qmail/control/badmailfrom/template-begin
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/qmail/control/badmailfrom/template-begin	2005-04-20 21:19:07.000000000 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/qmail/control/badmailfrom/template-begin	1969-12-31 17:00:00.000000000 -0700
@@ -1,2 +0,0 @@
-#spamer@spamdomain.com
-#@spamdomain.com
