--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList.spamhaus	2008-10-05 12:32:41.000000000 -0600
+++ smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList	2008-10-05 12:36:14.000000000 -0600
@@ -1 +1 @@
-bl.spamcop.net:combined.njabl.org:dnsbl.ahbl.org:dnsbl-1.uceprotect.net:dnsbl-2.uceprotect.net:list.dsbl.org:multihop.dsbl.org:psbl.surriel.com:sbl-xbl.spamhaus.org
+bl.spamcop.net:combined.njabl.org:dnsbl.ahbl.org:dnsbl-1.uceprotect.net:dnsbl-2.uceprotect.net:list.dsbl.org:multihop.dsbl.org:psbl.surriel.com:zen.spamhaus.org
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/migrate/20spamhaus.spamhaus	2008-10-05 12:36:38.000000000 -0600
+++ smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/migrate/20spamhaus	2008-10-05 12:41:40.000000000 -0600
@@ -0,0 +1,12 @@
+{
+    $qpsmtpd = $DB->get('qpsmtpd');
+    return unless $qpsmtpd;
+    return unless $qpsmtpd->prop('RBLList');
+
+    @rbl = split /[,:]/, $qpsmtpd->prop('RBLList');
+    if ( grep { /^sbl-xbl.spamhaus.org$/ } @rbl ) {
+        @rbl = grep { !/^sbl-xbl.spamhaus.org$/ } @rbl;
+        push @rbl, 'zen.spamhaus.org';
+        $qpsmtpd->set_prop('RBLList', join ':', @rbl);
+    }
+}
