diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/22dnsbl mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/22dnsbl
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/22dnsbl	2005-08-19 17:31:01.000000000 -0400
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/22dnsbl	2007-09-06 18:34:48.000000000 -0400
@@ -1,5 +1,5 @@
 {
     return "# dnsbl disabled" unless (${qpsmtpd}{DNSBL} eq "enabled");
 
-    "dnsbl";
+    "dnsbl disconnect";
 }
