diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/createlinks mezzanine_patched_smeserver-qpsmtpd-1.2.1/createlinks
--- smeserver-qpsmtpd-1.2.1/createlinks	2006-11-20 11:14:24.000000000 -0700
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/createlinks	2006-11-20 11:12:36.000000000 -0700
@@ -48,6 +48,7 @@
 		      ));
 
 my $control = "/var/qmail/control";
+templates2events("$control/rcpthosts", 'ip-change');
 templates2events("$control/badmailfrom", qw(
 			bootstrap-console-save
 			console-save
