--- /usr/share/qpsmtpd/plugins/disclaimer.orig	2008-02-27 10:34:17.000000000 +0100
+++ /usr/share/qpsmtpd/plugins/disclaimer	2008-06-19 16:19:43.000000000 +0200
@@ -10,6 +10,16 @@
   %{$self->{_args}} = @args;
 }
 
+sub read_disclaimer {
+  my ($self, $disclaimer_file) = @_;
+  my $disclaimer_dir = $self->{_args}->{disclaimer_dir} || "/service/qpsmtpd/config/";
+  return () unless open DISCLAIMER, "<$disclaimer_dir/$disclaimer_file";
+  my @lines = <DISCLAIMER>;
+  chomp @lines;
+  close DISCLAIMER;
+  return @lines;
+}
+
 sub hook_data_post {
   my ($self, $transaction) = @_;
 
@@ -45,10 +55,10 @@
 
   my $specific = 0;
   my @lines;
-  if ( @lines = $self->qp->config("disclaimer_$fromhost") ) {
+  if ( @lines = $self->read_disclaimer("disclaimer_$fromhost") ) {
     $self->log(LOGDEBUG, "Found disclaimer for domain: $fromhost");
     $specific = 1;
-  } elsif ( @lines = $self->qp->config("disclaimer") ) {
+  } elsif ( @lines = $self->read_disclaimer("disclaimer") ) {
     $self->log(LOGDEBUG, "Found general disclaimer");
   } else {
     $self->log(LOGERROR, "No disclaimer found for domain: $fromhost");
