diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/Bcc mezzanine_patched_smeserver-qpsmtpd-1.0.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/Bcc
--- smeserver-qpsmtpd-1.0.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/Bcc	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/Bcc	2006-01-25 11:16:01.013129158 +1100
@@ -0,0 +1 @@
+disabled
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/BccUser mezzanine_patched_smeserver-qpsmtpd-1.0.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/BccUser
--- smeserver-qpsmtpd-1.0.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/BccUser	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/BccUser	2006-01-25 11:16:05.894279892 +1100
@@ -0,0 +1 @@
+maillog
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/13bcc mezzanine_patched_smeserver-qpsmtpd-1.0.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/13bcc
--- smeserver-qpsmtpd-1.0.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/13bcc	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/13bcc	2006-01-25 11:16:47.638016951 +1100
@@ -0,0 +1,4 @@
+{
+     return "# bcc disabled" unless ($qpsmtpd{Bcc} eq "enabled");
+     return "bcc mode cc all " . $qpsmtpd{BccUser};
+}
