diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.0/root/var/service/sqpsmtpd/sqpsmtpd mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/var/service/sqpsmtpd/sqpsmtpd
--- smeserver-qpsmtpd-1.0.0/root/var/service/sqpsmtpd/sqpsmtpd	2005-08-19 15:31:03.000000000 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/var/service/sqpsmtpd/sqpsmtpd	2005-08-29 00:19:50.062426096 -0600
@@ -22,6 +22,6 @@
 #----------------------------------------------------------------------
 
 exec sslio -vv -/ ../qpsmtpd/ssl \
-	-U qpsmtpd \
+	-u qpsmtpd -U qpsmtpd \
 	/var/service/sqpsmtpd/qpsmtpd
 
