diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.2/createlinks mezzanine_patched_smeserver-qpsmtpd-1.0.2/createlinks
--- smeserver-qpsmtpd-1.0.2/createlinks	2006-02-17 17:11:38.000000000 -0500
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.2/createlinks	2006-02-17 17:11:08.000000000 -0500
@@ -76,6 +76,7 @@
 				badhelo
 				badrcptto_patterns
 				dnsbl_zones
+				dnsbl_allow
 				loglevel
 				memory_threshold
 				plugins
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/dnsbl_allow/80LocalNetworks mezzanine_patched_smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/dnsbl_allow/80LocalNetworks
--- smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/dnsbl_allow/80LocalNetworks	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/dnsbl_allow/80LocalNetworks	2006-02-17 17:10:39.000000000 -0500
@@ -0,0 +1,23 @@
+{
+    use esmith::util;
+
+    my @prefixes = esmith::util::computeAllLocalNetworkPrefixes($LocalIP,
+                                                                $LocalNetmask);
+
+    require esmith::NetworksDB;
+    my $n = esmith::NetworksDB->open;
+    foreach my $network ($n->get_all_by_prop(type => 'network'))
+    {
+        push(@prefixes,
+            esmith::util::computeAllLocalNetworkPrefixes(
+                $network->key, $network->prop('Mask')));
+    }
+
+    foreach my $prefix ( @prefixes )
+    {
+        my $dot = ( $prefix =~ /\d+\.\d+\.\d+\.\d+/ ) ? '' : '.';
+
+        $OUT .= $prefix . $dot . "\n";
+    }
+    $OUT .= "127.0.0.1";
+}
