diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.0/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList
--- smeserver-qpsmtpd-1.0.0/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList	2005-08-19 15:31:01.000000000 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList	2005-08-24 20:09:17.700704408 -0600
@@ -1 +1 @@
-sbl-xbl.spamhaus.org:whois.rfc-ignorant.org:dnsbl.njabl.org:relays.ordb.org
+sbl-xbl.spamhaus.org,whois.rfc-ignorant.org,dnsbl.njabl.org,relays.ordb.org
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.0/root/etc/e-smith/templates/var/service/qpsmtpd/config/dnsbl_zones/10RBLList mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/etc/e-smith/templates/var/service/qpsmtpd/config/dnsbl_zones/10RBLList
--- smeserver-qpsmtpd-1.0.0/root/etc/e-smith/templates/var/service/qpsmtpd/config/dnsbl_zones/10RBLList	2005-04-29 15:54:21.000000000 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/etc/e-smith/templates/var/service/qpsmtpd/config/dnsbl_zones/10RBLList	2005-08-24 20:09:07.344278824 -0600
@@ -1,5 +1,5 @@
 {
-    my @rbllist = split /:/, ${qpsmtpd}{RBLList} || '';
+    my @rbllist = split /,/, ${qpsmtpd}{RBLList} || '';
 
     return "# No RBLs are defined" unless (scalar @rbllist);
 
