diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.0/root/var/service/qpsmtpd/run mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/var/service/qpsmtpd/run
--- smeserver-qpsmtpd-1.0.0/root/var/service/qpsmtpd/run	2005-08-29 03:21:59.085259384 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/var/service/qpsmtpd/run	2005-08-29 03:19:53.400899520 -0600
@@ -26,7 +26,7 @@
 export QPSMTPD_CONFIG=/var/service/qpsmtpd/config
 
 exec /usr/local/bin/softlimit -m 25000000 \
-  ./qpsmtpd-forkserver \
+  /usr/bin/qpsmtpd-forkserver \
 	-u qpsmtpd \
 	-l 0.0.0.0 \
 	-p ${PORT:-25} \
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.0/root/var/service/sqpsmtpd/sqpsmtpd mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/var/service/sqpsmtpd/sqpsmtpd
--- smeserver-qpsmtpd-1.0.0/root/var/service/sqpsmtpd/sqpsmtpd	2005-08-29 03:21:58.963277928 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/var/service/sqpsmtpd/sqpsmtpd	2005-08-29 03:21:42.505779848 -0600
@@ -23,5 +23,5 @@
 
 exec sslio -vv -/ ../qpsmtpd/ssl \
 	-u qpsmtpd -U qpsmtpd \
-	/var/service/sqpsmtpd/qpsmtpd
+	/usr/bin/qpsmtpd
 
