diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.0/root/var/service/qpsmtpd/run mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/var/service/qpsmtpd/run
--- smeserver-qpsmtpd-1.0.0/root/var/service/qpsmtpd/run	2005-09-22 00:13:46.064847592 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/var/service/qpsmtpd/run	2005-09-22 00:10:40.840006048 -0600
@@ -25,7 +25,7 @@
 
 export QPSMTPD_CONFIG=/var/service/qpsmtpd/config
 
-exec /usr/local/bin/softlimit -m 25000000 \
+exec /usr/local/bin/softlimit -d 25000000 -s 25000000 -l 25000000 \
   /usr/bin/qpsmtpd-forkserver \
 	-u qpsmtpd \
 	-l 0.0.0.0 \
