diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.0/root/etc/e-smith/templates/var/qmail/control/badrcptto/20blockRoot mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/etc/e-smith/templates/var/qmail/control/badrcptto/20blockRoot
--- smeserver-qpsmtpd-1.0.0/root/etc/e-smith/templates/var/qmail/control/badrcptto/20blockRoot	2005-04-20 21:19:07.000000000 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.0/root/etc/e-smith/templates/var/qmail/control/badrcptto/20blockRoot	1969-12-31 17:00:00.000000000 -0700
@@ -1,16 +0,0 @@
-{
-    $OUT .= "root\n";
-    $OUT .= "root\@$DomainName\n";
-
-    foreach my $domain ( keys %Domains )
-    {
-        next unless $Domains{$domain}{type} eq "domain";
-        $OUT .= "root\@$domain\n";
-    }
-
-    foreach my $host ( keys %Hosts )
-    {
-        next unless $Hosts{$host}{type} eq "host";
-        $OUT .= "root\@$host\n";
-    }
-}
