diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates.metadata/var/service/qpsmtpd/config/plugins mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates.metadata/var/service/qpsmtpd/config/plugins
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates.metadata/var/service/qpsmtpd/config/plugins	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates.metadata/var/service/qpsmtpd/config/plugins	2006-11-17 13:05:48.000000000 -0700
@@ -0,0 +1 @@
+OUTPUT_FILENAME="/var/service/qpsmtpd/config/peers/0"
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates.metadata/var/service/qpsmtpd/peers/0 mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates.metadata/var/service/qpsmtpd/peers/0
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates.metadata/var/service/qpsmtpd/peers/0	2005-04-21 05:54:00.000000000 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates.metadata/var/service/qpsmtpd/peers/0	2006-11-17 13:04:43.000000000 -0700
@@ -1 +1,2 @@
+TEMPLATE_PATH="/var/service/qpsmtpd/config/plugins"
 PERMS=use esmith::ConfigDB; (esmith::ConfigDB->open_ro->get('qpsmtpd')->prop('access') eq "private") ? "000" : "0644"
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/var/service/qpsmtpd/config/plugins mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/var/service/qpsmtpd/config/plugins
--- smeserver-qpsmtpd-1.2.1/root/var/service/qpsmtpd/config/plugins	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/var/service/qpsmtpd/config/plugins	2006-11-17 13:10:14.000000000 -0700
@@ -0,0 +1,6 @@
+#------------------------------------------------------------
+#              !!DO NOT MODIFY THIS FILE!!
+#
+# Plugins are now loaded by the peers plugin from config/peers/*
+#------------------------------------------------------------
+peers
