diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/BccMode mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/BccMode
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/BccMode	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/BccMode	2006-08-25 01:30:34.080443489 +1000
@@ -0,0 +1 @@
+cc
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/13bcc mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/13bcc
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/13bcc	2006-02-06 01:38:30.000000000 +1100
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/13bcc	2006-08-25 01:30:34.081443315 +1000
@@ -1,4 +1,4 @@
 {
      return "# bcc disabled" unless ($qpsmtpd{Bcc} eq "enabled");
-     return "bcc mode cc all " . $qpsmtpd{BccUser};
+     return "bcc mode " . $qpsmtpd{BccMode} . " all " . $qpsmtpd{BccUser};
 }
