diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.0/createlinks mezzanine_patched_smeserver-qpsmtpd-1.2.0/createlinks
--- smeserver-qpsmtpd-1.2.0/createlinks	2006-03-15 07:09:01.000000000 +1100
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.0/createlinks	2006-06-21 01:57:19.703066508 +1000
@@ -105,6 +105,9 @@
 				pseudonym-create
 				pseudonym-delete
 				pseudonym-modify
+				group-create
+				group-delete
+				group-modify
 		));
 
 for my $daemon ( qw(qpsmtpd sqpsmtpd) )
