diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList	2005-10-06 19:21:58.000000000 -0600
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/RBLList	2007-01-10 16:39:07.000000000 -0700
@@ -1 +1 @@
-sbl-xbl.spamhaus.org,whois.rfc-ignorant.org,dnsbl.njabl.org,relays.ordb.org
+bl.spamcop.net:combined.njabl.org:dnsbl.ahbl.org:dnsbl-1.uceprotect.net:dnsbl-2.uceprotect.net:list.dsbl.org:multihop.dsbl.org:psbl.surriel.com:sbl-xbl.spamhaus.org
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/SBLList mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/SBLList
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/SBLList	2006-02-05 07:38:30.000000000 -0700
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/defaults/qpsmtpd/SBLList	2007-01-10 16:39:24.000000000 -0700
@@ -1 +1 @@
-dsn.rfc-ignorant.org
+bogusmx.rfc-ignorant.org:multi.surbl.org:black.uribl.com:rhsbl.sorbs.net:bulk.rhs.mailpolice.com:fraud.rhs.mailpolice.com:porn.rhs.mailpolice.com:adult.rhs.mailpolice.com:ex.dnsbl.org:blackhole.securitysage.com
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/migrate/20ordb.org mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/migrate/20ordb.org
--- smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/migrate/20ordb.org	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.1/root/etc/e-smith/db/configuration/migrate/20ordb.org	2007-01-10 18:05:30.000000000 -0700
@@ -0,0 +1,7 @@
+{
+    $qpsmtpd = $DB->get('qpsmtpd');
+    return unless $qpsmtpd;
+
+    @rbl = grep { ! /ordb.org/ } split /[,:]/, $qpsmtpd->prop('RBLList');
+    $qpsmtpd->set_prop('RBLList', join ':', @rbl);
+}
