diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.2/createlinks mezzanine_patched_smeserver-qpsmtpd-1.0.2/createlinks
--- smeserver-qpsmtpd-1.0.2/createlinks	2006-02-05 09:38:30.000000000 -0500
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.2/createlinks	2006-02-06 21:09:31.000000000 -0500
@@ -88,7 +88,11 @@
 				invalid_resolvable_fromhost
 				));
 templates2events("$config/badhelo", 'ip-change');
-
+templates2events("$config/whitelistsenders", qw(
+				bootstrap-console-save
+				domain-create
+				domain-delete
+				));
 templates2events("$config/goodrcptto", qw(
 				bootstrap-console-save
 				email-update
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/15whitelist_soft mezzanine_patched_smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/15whitelist_soft
--- smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/15whitelist_soft	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/plugins/15whitelist_soft	2006-02-05 21:44:49.000000000 -0500
@@ -0,0 +1,3 @@
+{
+    return ($qpsmtpd{RequireResolvableFromHost} eq 'yes') ? "whitelist_soft" : "";
+}
diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/whitelistsenders/50domains mezzanine_patched_smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/whitelistsenders/50domains
--- smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/whitelistsenders/50domains	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_smeserver-qpsmtpd-1.0.2/root/etc/e-smith/templates/var/service/qpsmtpd/config/whitelistsenders/50domains	2006-02-06 14:54:11.000000000 -0500
@@ -0,0 +1,4 @@
+{
+    use esmith::DomainsDB;
+    $OUT = join "\n", map { $_->key } esmith::DomainsDB->open_ro->domains;
+}
