diff -Nur -x '*.orig' -x '*.rej' smeserver-qpsmtpd-1.2.0/createlinks mezzanine_patched_smeserver-qpsmtpd-1.2.0/createlinks
--- smeserver-qpsmtpd-1.2.0/createlinks	2006-06-26 20:44:29.200809763 +1000
+++ mezzanine_patched_smeserver-qpsmtpd-1.2.0/createlinks	2006-06-26 20:43:16.772821097 +1000
@@ -60,6 +60,9 @@
 			user-create
 			user-delete
 			user-modify
+			group-create
+			group-delete
+			group-modify
 			pseudonym-create
 			pseudonym-delete
 			pseudonym-modify
